luigi/contrib/kubernetes.py

```diff
@@ -42,74 +44,46 @@
 logger = logging.getLogger('luigi-interface')
 
 try:
-    from pykube.config import KubeConfig
-    from pykube.http import HTTPClient
-    from pykube.objects import Job, Pod
-except ImportError:
-    logger.warning('pykube is not installed. KubernetesJobTask requires pykube.')
+    import kubernetes as kubernetes_api
+except ImportError as i:
+    logger.warning("WARNING: kubernetes is not installed. KubernetesJobTask requires kubernetes")
+    logger.warning("  Please run 'pip install kubernetes' and try again")
 
 
 class kubernetes(luigi.Config):
-    auth_method = luigi.Parameter(
-        default="kubeconfig",
-        description="Authorization method to access the cluster")
-    kubeconfig_path = luigi.Parameter(
-        default="~/.kube/config",
-        description="Path to kubeconfig file for cluster authentication")
-    max_retrials = luigi.IntParameter(
-        default=0,
-        description="Max retrials in event of job failure")
     kubernetes_namespace = luigi.OptionalParameter(
         default=None,
         description="K8s namespace in which the job will run")
 
-
 class KubernetesJobTask(luigi.Task):
     __DEFAULT_POLL_INTERVAL = 5  # see __track_job
     __DEFAULT_POD_CREATION_INTERVAL = 5
-    _kubernetes_config = None  # Needs to be loaded at runtime
 
     def _init_kubernetes(self):
         self.__logger = logger
-        self.__logger.debug("Kubernetes auth method: " + self.auth_method)
-        if self.auth_method == "kubeconfig":
-            self.__kube_api = HTTPClient(KubeConfig.from_file(self.kubeconfig_path))
-        elif self.auth_method == "service-account":
-            self.__kube_api = HTTPClient(KubeConfig.from_service_account())
-        else:
-            raise ValueError("Illegal auth_method")
+        self.__kubernetes_core_api = kubernetes_api.client.CoreV1Api()
+        # Configs can be set in Configuration class directly or using helper utility, by default lets try to load in-cluster config
+        # TODO: Make library support forcing one of these instead of automatic cascading logic...?
+        try:
+           kubernetes_api.config.load_incluster_config()
+        except Exception as e:
+           try:
+              kubernetes_api.config.load_kube_config()
+           except Exception as ex:
+              raise ex
```

Comment:
Note for reviewers: This is the "new" cascading automated configuration handling.  If we're in Kubernetes, it will automatically detect this and use in-cluster configuration and API endpoints, if that fails, it will attempt to use the kubeconfig file and env vars that help support it.